test/jdk/java/awt/TrayIcon/DisposeInActionEventTest/DisposeInActionEventTest.java

```diff
@@ -21,76 +21,69 @@
-                    "  action event. Brief information about action events is printed",
-                    "  below. After each action event the tray icon is removed from",
-                    "  the tray and then added back in a second.",
-                    "The test performs some automatic checks when removing the icon. If",
-                    "  something is wrong the corresponding message is displayed below.",
-                    "  Repeat double-clicks several times. If no 'Test FAILED' messages",
-                    "  are printed, press PASS button else FAIL button."
-            };
+    public static void main(String[] args) throws Exception {
+        if (!SystemTray.isSupported()) {
+            throw new jtreg.SkippedException("The test cannot be run because " +
+                    "SystemTray is not supported.");
         }
-        Sysout.createDialogWithInstructions(instructions);
-    }
+        String clickInstruction = (Platform.isOSX()) ? "Right-click" : "Double-left click";
+
+        String instructions = "When the test starts, it adds the icon to the tray area. If you\n" +
+                       "  don't see a tray icon, please, make sure that the tray area\n" +
+                       "  (also called Taskbar Status Area on MS Windows, Notification\n" +
+                       "  Area on Gnome or System Tray on KDE) is visible.\n\n" +
+                        clickInstruction + " the button on the tray icon to trigger the\n" +
+                       "  action event. Brief information about action events is printed\n" +
+                       "  in the frame. After each action event the tray icon is removed from\n" +
+                       "  the tray and then added back in a second.\n\n" +
+                       "The test performs some automatic checks when removing the icon. If\n" +
+                       "  something is wrong the corresponding message is displayed below.\n" +
+                       "  Repeat clicks several times. If no 'Test FAILED' messages\n" +
+                       "  are printed, press PASS button else FAIL button.";
 
-    @Override
-    public void start() {
-        setSize(200, 200);
-        setVisible(true);
-        validate();
+        PassFailJFrame.builder()
+                .title("Test Instructions Frame")
+                .instructions(instructions)
+                .testTimeOut(10)
+                .rows(15)
+                .columns(45)
+                .testUI(DisposeInActionEventTest::showFrameAndIcon)
+                .build()
+                .awaitAndCheck();
+    }
 
-        if (!traySupported) return;
+    private static JFrame showFrameAndIcon() {
+        JFrame frame = new JFrame("DisposeInActionEventTest");
+        frame.setLayout(new BorderLayout());
 
-        System.setProperty("sun.awt.exception.handler", "DisposeInActionEventTest$EDTExceptionHandler");
+        textArea = new JTextArea();
+        frame.getContentPane().add(textArea);
```

Comment:
```suggestion
        frame.getContentPane().add(new JScrollPane(textArea));
```
Put the text area into a scroll pane to allow viewing all text even it doesn't fit. And the messages with `ActionEvent` do not fit.

Reply:
@alisenchung The ActionEvent messages are not seen in the "Event Message Display" window with the current changes. 
The textArea should be added to JScrollPane as @aivanov-jdk suggested and not as `spane.add(textArea)`.